File: repository/src/main/java/by/rekuts/travelagency/repository/TourRepository.java
package by.rekuts.travelagency.repository;

import by.rekuts.travelagency.domain.Tour;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.JpaSpecificationExecutor;

/**
 * Interface TourService concerns to DAO layer.
 * Realisations of this interface do some CRUD operations with Tour objects in database.
 */
public interface TourRepository extends JpaRepository<Tour, Integer>, JpaSpecificationExecutor<Tour> {
}

File: repository/src/main/scala/by/rekuts/travelagency/repository/TourRepository.scala
package by.rekuts.travelagency.repository

import by.rekuts.travelagency.domain.Tour

/**
  * Trait TourService concerns to DAO layer.
  * Realisations of this trait do some CRUD operations with Tour objects in database.
  */
trait TourRepository extends CrudRepository[Tour]